lib/aws-record/record/item_collection.rb

```diff
@@ -91,9 +92,16 @@ def empty?
       def _build_items_from_response(items, model)
         ret = []
         items.each do |item|
-          record = model.new
+          model_class = model
+          if @model_filter
+            model_class = @model_filter.call(item)
+            next unless model_class # skip nils
+            record = model_class.new
+          else
+            record = model.new
+          end
```

Comment:
A couple alternatives -- if you dislike using ternaries, you could just factor out the `.new`:

```
model_class = model
if @model_filter
  model_class = @model_filter.call(item)
  next unless model_class
end
record = model_class.new
```

If you're fine with ternaries:

```
model_class = @model_filter ? @model_filter.call(item) : model
next unless model_class
record = model_class.new
```

(The latter assumes the supplied `model` is not `nil`. But it looks like `model` is always `@model`. So you could prevent that case by checking for it in the constructor.)

Reply:
I think the ternary here is concise and clear.  I've cleaned up following this - good suggestion.